### ISSUE TITLE
Singular resources

### ISSUE THREAD

**Issue by oliverbarnes** [pixelhandler/ember-jsonapi-resources#26]
Hi again Bill, 

So I'm trying to get a `me` resource going, adapting a `/me` endpoint on the API returning the current user.

I generated a resource with `ember g resource me`, and singularized all references to 'mes' into 'me' on the intializer and service/model/adapter/serializer files. Here's the initializer:

```
import Service from '../services/me';
import Model from '../models/me';
import Adapter from '../adapters/me';
import Serializer from '../serializers/me';

export function initialize(container, application) {
  const adapter = 'service:me-adapter';
  const serializer = 'service:me-serializer';
  const service = 'service:me';
  const model = 'model:me';

  application.register(model, Model, { instantiate: false, singleton: false });
  application.register(service, Service);
  application.register(adapter, Adapter);
  application.register(serializer, Serializer);

  application.inject('service:store', 'me', service);
  application.inject(service, 'serializer', serializer);
}

export default {
  name: 'me-service',
  after: 'store',
  initialize: initialize
};
```

And the adapter:

```
import ApplicationAdapter from './application';
import config from 'client/config/environment';

export default ApplicationAdapter.extend({
  type: 'me',

  url: config.APP.API_HOST + 'me'
});
```

On the route I'm trying to fetch it in the model like this:

```
import Ember from 'ember';
import AuthenticatedRouteMixin from 'simple-auth/mixins/authenticated-route-mixin';

export default Ember.Route.extend(AuthenticatedRouteMixin, {
  model() {
    return this.store.find('me', {});
  }
});
```

But the store still looks for a 'mes' service:

```
Error while processing route: dashboard mes service not initialized Error: mes service not initialized
```

Not sure if this is a bug or just not implemented yet (given `JSONAPI::Resources` has a `json_resource` for singleton resources, I figured this would be supported in EJR as well), or whether I'm missing something. Let me know also if this is more of a SO or slack question really... =D

**Comment by pixelhandler** [comment]
@oliverbarnes in the examples on the jsonapi.org format I only saw plural endpoints and types. I just looked over the spec and found:

> Note: This spec is agnostic about inflection rules, so the value of type can be either plural or singular. However, the same value should be used consistently throughout an implementation.

So it seems like a singular name for a type and endpoint fit the spec. Perhaps the less common approach. I actually didn't know JSONAPI::Resources supported singular resources.

I'll look over your PR soon as I can.
